chemprop/cli/train.py

```diff
@@ -636,6 +639,83 @@ def build_splits(args, format_kwargs, featurization_kwargs):
     return train_data, val_data, test_data
 
 
+def summarize(args, dataset: _MolGraphDatasetMixin) -> tuple[list, list]:
+    if args.task_type in ["regression", "regression-mve", "regression-evidential"]:
+        if isinstance(dataset, MulticomponentDataset):
+            y = dataset.datasets[0].Y
+        else:
+            y = dataset.Y
+        y_mean = np.nanmean(y, axis=0)
+        print("YMEAN", y_mean)
+        y_std = np.nanstd(y, axis=0)
+        y_median = np.nanmedian(y, axis=0)
+        mean_dev_abs = np.abs(y - y_mean)
+        frac_1_sigma = np.sum((mean_dev_abs < y_std), axis=0) / len(y)
+        frac_2_sigma = np.sum((mean_dev_abs < 2 * y_std), axis=0) / len(y)
+
+        column_headers = ["Statistic"] + [f"Value {i}" for i in range(y.shape[1])]
+        table_rows = [
+            ["Num. datapoints"] + [f"{len(y)}" for i in range(y.shape[1])],
+            ["Mean"] + [f"{mean:0.2f}" for mean in y_mean],
+            ["Std. dev."] + [f"{std:0.2f}" for std in y_std],
+            ["Median"] + [f"{median:0.2f}" for median in y_median],
+            ["% within 1 s.d."] + [f"{sigma:0.0%}" for sigma in frac_1_sigma],
+            ["% within 2 s.d."] + [f"{sigma:0.0%}" for sigma in frac_2_sigma],
+        ]
+        return (column_headers, table_rows)
+    elif args.task_type in [
+        "classification",
+        "classification-dirichlet",
+        "multiclass",
+        "multiclass-dirichlet",
+    ]:
+        if isinstance(dataset, MulticomponentDataset):
+            y = dataset.datasets[0].Y
+        else:
+            y = dataset.Y
+
+        mask = np.isnan(y)
+        classes = np.sort(np.unique(y[~mask]))
+
+        class_counts = np.stack([(classes[:, None] == y[:, i]).sum(1) for i in range(y.shape[1])])
+        class_fracs = class_counts / y.shape[0]
+        nan_count = np.nansum(mask, axis=0)
+        nan_frac = nan_count / len(y)
+
+        column_headers = ["Class"] + [f"Count/Percent {i}" for i in range(y.shape[1])]
```

Comment:
It would be nice to use the actual target names instead of 0, 1, 2, etc. if we can. If you look at `evaluate_and_save_predictions` in `train.py` there is an example of how to get the `target_cols`. It isn't very elegant and we should probably find a different solution in the long run, but I feel that will work for now.

Reply:
Thanks for pointing me toward the example, Done!